### PR TITLE
setup.py: Fail gracefully when NumPy is not installed

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Cant we just have a try catch as the first line?

**Comment by stefanv** [comment]
Well, no, there are cases in which we don't need NumPy.

**Comment by hmaarrfk** [comment]
Have users recently complained about the inability to install nunpy?

**Comment by stefanv** [comment]
No, this is for the case where they try to install skimage and numpy is missing, so that they get a graceful error message.

**Comment by hmaarrfk** [comment]
Will this automatically install numpy in time to build exceptions?

**Comment by stefanv** [comment]
No, as far as I know that cannot be done.

File: setup.py [hunk 1]
```diff
@@ -33,8 +33,6 @@
 from distutils.command.sdist import sdist
 from distutils.errors import CompileError, LinkError
 
-from numpy.distutils.command.build_ext import build_ext
```
Comment (hmaarrfk) [inline setup.py]: Why not have `try: .... Except ImportError: raise RuntimeError('numpy needs to be installed to install scikit-image')`

**Comment by stefanv** [comment]
Because some setup.py commands don't require numpy at all, like egg info.

**Comment by jni** [comment]
Any fixes to the setup.py can come in later PRs. This solves a reported problem and passes CI, so in it goes!

I suspect most of these hacks will get fixed when we go back to pyproject.toml. I think the time for giving that another go is coming soon.

**Pull request merged**